app/objects/secondclass/c_host.py

```diff
@@ -8,15 +8,31 @@
 
 
 class HostAccess(Enum):
+
     DENY = -1
     STANDARD = 0
     ALLOW = 1
 
 
 class AbilitySchema(ma.Schema):
+
     uuid = ma.fields.String()
     success_prob = ma.fields.Number()
 
+    @ma.post_load()
+    def build_ability(self, data, **_):
+        return Ability(**data)
+
+
+class Ability(BaseObject):
```

Comment:
Abilities must inherit from `BaseObject` in order for the data service to serialize and de-serialize them.